Add Group Name field

stenvdb opened the pull request:
For a project we use, we needed the ability to update the group name in Campaign Monitor so we can better filter out some emails. I've updated the adapter for the optional extra field. It fallbacks to the default name (which was just 'Craft CMS').

stenvdb committed:
Add Group Name field

heymarkreeves commented:
Hi, @stenvdb!

Any chance you've tested what happens if you update the plugin but have not yet re-saved its settings with the Group Name value set? I see where there's a fallback value [here](https://github.com/clearbold/craft-campaignmonitor-transactional/blob/632195df898a2a14ea944daff70b97527b121601/src/CmTransactionalTransport.php), but not here:

```
$groupName = Craft::parseEnv($this->groupName);
```

I'm just wondering what happens if someone updates the plugin without re-saving its settings to set that value. I'll need to test that.

Thanks!

stenvdb committed:
Catch unsaved mail adapter settings

stenvdb commented:
Good catch, it did not work 🙈 I've added a better conditional and removed the argument type since it can be null if the settings haven't been updated.
